privacyidea/api/user.py

```diff
@@ -176,7 +176,8 @@ def create_user_api():
     # Remove the password from the attributes, so that we can hide it in the
     # logs
     password = attributes.get("password")
-    del attributes["password"]
+    if "password" in attributes:
+        del attributes["password"]
```

Comment:
is there a reason why we delete the pw from the attributes here and add it again in [`create_user()`](https://github.com/privacyidea/privacyidea/blob/1143f41d7abc9aa6809fd50ae91d86b4eb6a071f/privacyidea/lib/user.py#L511)?

Reply:
Yes. Otherwise we can not hide the password from the @log wrapper.
It would be contained in the params dictionary and would be visible in the log.debug level.

The create_user is decorated with
~~~~python
@log_with(log, hide_kwargs=["password"])
def create_user(resolvername, attributes, password=None):
~~~~
Which means, never show the arg "password" in the logs :-)